tools/python/find_optimizer_opset_version_updates_required.py

```diff
@@ -96,13 +96,22 @@ def get_multiline_call_args_from_file(filename: str, function_or_declaration: st
     return results
 
 
-def get_latest_op_versions(root_dir):
+def _add_if_newer(domain: str, op: str, opset: int, op_to_opset: typing.Dict[str, int]):
+    key = domain + "." + op
+    if key not in op_to_opset or op_to_opset[key] < opset:
+        op_to_opset[key] = opset
+
+
+def get_latest_ort_op_versions(root_dir):
     """Find the entries for the latest opset for each operator."""
 
     op_to_opset = {}
     files = [
-        os.path.join(root_dir, "onnxruntime/core/providers/cpu/cpu_execution_provider.cc"),
+        # for ONNX operators we use get_latest_onnx_op_versions
+        # os.path.join(root_dir, "onnxruntime/core/providers/cpu/cpu_execution_provider.cc"),
+        # for internal kernels we use the current registrations
         os.path.join(root_dir, "onnxruntime/contrib_ops/cpu/cpu_contrib_kernels.cc"),
+        os.path.join(root_dir, "onnxruntime/contrib_ops/cuda/cuda_contrib_kernels.cc"),
```

Comment:
We have a contrib op that only has a CUDA version so need to check this as well.